core/src/main/java/hudson/FilePath.java

```diff
@@ -1654,11 +1690,36 @@ public String invoke(File dir, VirtualChannel channel) throws IOException {
                 throw new IOException("Failed to create a temporary directory in " + dir, e);
             }
 
+            boolean lossy = false;
+            try {
+                losslessWrite(f);
+            } catch (CharacterCodingException e) {
+                Functions.printStackTrace(e, listener.error(
+                        "Lossless encoding in JVM default character set "
+                                + Charset.defaultCharset().name()
+                                + " failed. Falling back to lossy encoding."));
+                lossy = true;
+            }
+            if (lossy) {
+                lossyWrite(f);
```

Comment:
```suggestion
            try {
                losslessWrite(f);
            } catch (CharacterCodingException e) {
                Functions.printStackTrace(e, listener.error(
                        "Lossless encoding in JVM default character set "
                                + Charset.defaultCharset().name()
                                + " failed. Falling back to lossy encoding."));
                lossyWrite(f);
```

?

Reply:
A request made without providing reasoning.

Reply:
Just code simplification, unless I am missing some reason for the variable.